### PR TITLE
Guess who's back - Java 8 is back again - tell a friend 

### PR DESCRIPTION
**Pull request by ohadbitt**
Also updating bom version and msal

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -8,14 +8,17 @@ on:
...
+    strategy:
+      matrix:
+        java: [ '8' ]
+    name: Java ${{ matrix.java }}
     steps:
       - uses: actions/checkout@v2
-      - name: Setup java 11
+      - name: Setup java 8
         uses: actions/setup-java@v3
         with:
```
Comment (AsafMah) [inline .github/workflows/release.yml:19]: ${{ matrix.java }} if you're gonna use it

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 language: java
 jdk:
-  - oraclejdk11
+  - oraclejdk8
```
Comment (AsafMah) [inline .travis.yml:3]: I don't think we need this file at all anymore (we don't use travisci)

File: pom.xml [hunk 3]
```diff
@@ -32,19 +32,19 @@
     </developers>
 
     <properties>
-        <revision>4.0.4</revision> <!-- CHANGE THIS to adjust project version-->
+        <revision>4.0.5</revision> <!-- CHANGE THIS to adjust project version-->
```
Comment (AsafMah) [inline pom.xml:35]: This is atleast a 4.1.0 no?

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java [hunk 4]
```diff
@@ -16,13 +16,13 @@
 public class ClientDetails {
 
     public static final String NONE = "[none]";
-    private static ConcurrentInitializer<String> defaultApplication = new LazyInitializer<>() {
+    private static ConcurrentInitializer<String> defaultApplication = new LazyInitializer() {
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java:19]: Pretty sure <> also works in java 8

**Review by AsafMah — COMMENTED** [review]
Nothing major

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java (hunk 4 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientDetails.java:19]: it fails

File: pom.xml (hunk 3 above)
Comment (ohadbitt) [reply pom.xml:35]: 5.0.0

**Pull request merged**